src/V3Const.cpp

```diff
@@ -1346,15 +1346,27 @@ class ConstVisitor final : public AstNVisitor {
                        && (nodep->msbConst() > maxDeclBit || nodep->lsbConst() > maxDeclBit)) {
                 // See also warning in V3Width
                 // Must adjust by element width as declRange() is in number of elements
-                nodep->v3warn(SELRANGE,
-                              "Selection index out of range: "
-                                  << (nodep->msbConst() / nodep->declElWidth()) << ":"
-                                  << (nodep->lsbConst() / nodep->declElWidth()) << " outside "
-                                  << nodep->declRange().hiMaxSelect() << ":0"
-                                  << (nodep->declRange().lo() >= 0
-                                          ? ""
-                                          : (" (adjusted +" + cvtToStr(-nodep->declRange().lo())
-                                             + " to account for negative lsb)")));
+                if (nodep->declElWidth() == 0) {
+                    nodep->v3warn(SELRANGE,
+                                "Selection index out of range (nodep->declElWidth == 0): "
+                                    << nodep->msbConst() << ":"
+                                    << nodep->lsbConst() << " outside "
```

Comment:
```suggestion
                                "Selection index out of range (with declared width 0): "
                                    << nodep->msbConst() << ":"
                                    << nodep->lsbConst() << " outside "
```

Also to reduce duplication I'd suggest create a variable with the msb and lsb to print, and conditionalize the "  (with declared width 0)".  That way the mostly duplicated warning code can go away.